core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -748,25 +739,30 @@ func (r *EvmRegistry) checkUpkeeps(ctx context.Context, keys []ocr2keepers.Upkee
 }
 
 // TODO (AUTO-2013): Have better error handling to not return nil results in case of partial errors
-func (r *EvmRegistry) simulatePerformUpkeeps(ctx context.Context, checkResults []EVMAutomationUpkeepResult21) ([]EVMAutomationUpkeepResult21, error) {
+func (r *EvmRegistry) simulatePerformUpkeeps(ctx context.Context, checkResults []ocr2keepers.CheckResult) ([]ocr2keepers.CheckResult, error) {
 	var (
 		performReqs     = make([]rpc.BatchElem, 0, len(checkResults))
 		performResults  = make([]*string, 0, len(checkResults))
 		performToKeyIdx = make([]int, 0, len(checkResults))
 	)
 
-	for i, checkResult := range checkResults {
-		if !checkResult.Eligible {
+	for i, cr := range checkResults {
+		if !cr.Eligible {
 			continue
 		}
 
-		opts, err := r.buildCallOpts(ctx, big.NewInt(int64(checkResult.Block)))
+		block, upkeepId, err := r.getCheckBlockAndUpkeepId(cr.Payload)
+		if err != nil {
+			return nil, err
+		}
+
+		opts, err := r.buildCallOpts(ctx, block)
```

Comment:
simulate at the check block (NOT the trigger block)

Reply:
I talked to Ryan and we concluded they are the same thing now. The block number in the `Trigger` is the block to run all pipeline check on.